include/revng/Pipeline/Global.h

```diff
@@ -36,6 +36,8 @@ class Global {
   virtual llvm::Error applyDiff(const llvm::MemoryBuffer &Diff) = 0;
   virtual llvm::Error serialize(llvm::raw_ostream &OS) const = 0;
   virtual llvm::Error deserialize(const llvm::MemoryBuffer &Buffer) = 0;
+  virtual llvm::Error verify() const = 0;
+  virtual llvm::Error verifyOther(const llvm::MemoryBuffer &Buffer) const = 0;
```

Comment:
Basically it's `verify` but instead of verifying itself it will verify if what's in the buffer, this is used to have a dry-run `set_global`

Reply:
Mmmmh. Can you make it a static method?
If it doesn't use `this`, it doesn't make sense to make a method.

Reply:
It has to be `virtual` because each `Global` has its own specialized `verify` (and consequently `verifyOther`)

Reply:
This means that a `Global` represenst at the same time the type of a global and the data of that global, which is not super nice. They could be decoupled.

In order to avoid proliferation of `xOther` methods, I suggest we introduce some method such as `std::unique_ptr<Global> createNewOfSameType(MemoryBuffer)` and then you can call `verify` on that one.

Reply:
Done, now there's a `createNew` function
It also helped remove a couple of double deserializations, which is nice